include/autoppl/expression/distribution/base.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+#include <cmath>
+#include <numeric>
+
+namespace ppl {
+namespace expr {
+
+/*
+ * The Base objects contain static member functions 
+ * that compute pdf and log_pdf.
+ * These are useful stand-alone functions and the distribution objects
+ * such as Uniform and Normal simply wrap these functions.
+ */
+
+/*
+ * Continuous Distributions
+ */
+
+struct UniformBase
```

Comment:
Why do we need this Base class? Is this just to allow these to be static member functions? This, along with the calling wrapper, has a kind of code-smell to me. We're splitting any new distribution we write across 3 files with lots of boilerplate.

Reply:
My thinking was that the Uniform, Normal, Bernoulli classes are to represent an expression in the ultimate big model expression. The expression is only there to give structure to the model and that's its only purpose. And these functions can exist without these classes. I can see other parts of the library needing to compute a pdf of a known distribution - shouldn't have to construct an expression just to do that.